fiovb/ta/entry.c

```diff
@@ -160,6 +162,89 @@ static TEE_Result delete_value(char *name, size_t name_sz)
 	return res;
 }
 
+static bool is_rollback_protected(void)
+{
+	TEE_Result res = TEE_SUCCESS;
+	uint32_t value_sz = MAX_SIMPLE_VALUE_SIZE;
+	char value[MAX_SIMPLE_VALUE_SIZE];
+	uint32_t count;
+
+	res = read_value(ROLLBACK_PROT, strlen(ROLLBACK_PROT) + 1,
+			value, value_sz, &count);
+	if (res == TEE_SUCCESS) {
+		IMSG("Found %s, rollback protection is enabled",
+		      ROLLBACK_PROT);
+		return true;
+
+	} else if (res == TEE_ERROR_ITEM_NOT_FOUND) {
+		IMSG("Not found %s, rollback protection OTP is not blown",
+		      ROLLBACK_PROT);
+		return false;
+	} else {
+		EMSG("Can't read named object '%s' value, res = 0x%x",
+		     ROLLBACK_PROT, res);
+	}
+
+	return false;
+}
+
+static bool is_version_incremental(char *new_ver_str,
+				   uint32_t new_ver_sz)
+{
+	TEE_Result res = TEE_SUCCESS;
+	char value[MAX_SIMPLE_VALUE_SIZE];
+	uint32_t count;
+	uint64_t current_ver, new_ver;
+
+	res = read_value(BOOTFIRM_VER, strlen(BOOTFIRM_VER) + 1,
+			value, MAX_SIMPLE_VALUE_SIZE, &count);
+	if (res == TEE_ERROR_ITEM_NOT_FOUND) {
+		IMSG("Not found %s, writing firmware version first time",
+		     BOOTFIRM_VER);
+		return true;
+	} else if (res == TEE_SUCCESS) {
```

Comment:
I'd remove the else and start with just a single if since we return in the previous if